diff --git a/src/lxappearance.c b/src/lxappearance.c
index a4b7db3..337548a 100644
--- a/src/lxappearance.c
+++ b/src/lxappearance.c
@@ -243,6 +243,19 @@ static void lxappearance_save_gtkrc()
     };
 
     char **gtkrc_files = gtk_rc_get_default_files();
+    int GTK2_RC_FILES_SET = 0;
+    if (gtkrc_files == NULL) {
+        const char *tmpEnv = g_getenv("GTK2_RC_FILES");
+        if(tmpEnv) {
+            GTK2_RC_FILES_SET = 1;
+            char *env = malloc(sizeof(char) * strlen(tmpEnv) + 1);
+            strcpy(env, tmpEnv);
+            gtkrc_files = malloc(sizeof(char*) * 2);
+            *gtkrc_files = env;
+            *(gtkrc_files + 1) = NULL;
+            g_debug("GTK2_RC_FILES set to: %s", *gtkrc_files);
+        }
+    }
     char *file_path = NULL;
     GString* content = g_string_sized_new(512);
 
@@ -261,15 +274,15 @@ static void lxappearance_save_gtkrc()
         g_debug("none found, use hardcoded %s/.gtkrc-2.0",g_get_home_dir());
         file_path = g_build_filename(g_get_home_dir(), ".gtkrc-2.0", NULL);
     }
-    g_string_append(content,
-        "# DO NOT EDIT! This file will be overwritten by LXAppearance.\n"
-        "# Any customization should be done in ~/.gtkrc-2.0.mine instead.\n\n");
+    // g_string_append(content,
+    //     "# DO NOT EDIT! This file will be overwritten by LXAppearance.\n"
+    //     "# Any customization should be done in ~/.gtkrc-2.0.mine instead.\n\n");
 
     /* include ~/.gtkrc-2.0.mine first to be able to apply changes done
        by LXAppearance if the same settings exist in that file */
-    g_string_append_printf(content,
-        "include \"%s/.gtkrc-2.0.mine\"\n",
-        g_get_home_dir());
+    // g_string_append_printf(content,
+    //     "include \"%s/.gtkrc-2.0.mine\"\n",
+    //     g_get_home_dir());
 
     if(app.widget_theme)
         g_string_append_printf(content,
@@ -435,6 +448,10 @@ static void lxappearance_save_gtkrc()
     g_string_free(content, TRUE);
     g_key_file_free(content_gtk3);
     g_free(file_path);
+    if (GTK2_RC_FILES_SET) {
+        free(*gtkrc_files);
+        free(gtkrc_files);
+    }
 }
 
 static void lxappearance_save_lxsession()
